daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/RefTest.scala

```diff
@@ -183,6 +183,55 @@ class RefTest extends AnyFreeSpec with Matchers with TableDrivenPropertyChecks w
     }
   }
 
+  "PackageVersion.fromString" - {
+    "parse valid package version strings" in {
+      val testCases = Table(
+        "Raw version string" -> "Parsed package version",
+        "0" -> ImmArray(0),
+        "1" -> ImmArray(1),
+        "1.10" -> ImmArray(1, 10),
+        Int.MaxValue.toString -> ImmArray(Int.MaxValue),
+        "1.0.2" -> ImmArray(1, 0, 2),
+        "0" + ".123" * 255 + ".12" -> ImmArray.from(Seq(0) ++ Seq.fill(255)(123) ++ Seq(12)),
+      )
+
+      testCases.forEvery { (input, expected) =>
+        PackageVersion.fromString(input) shouldBe Right(PackageVersion(expected))
+      }
+    }
+
+    "reject invalid package version strings" in {
+      val testCases = Table(
+        "(Invalid) Raw version string",
+        "",
+        ".",
+        "1.",
+        ".0",
+        "00",
+        "1.002",
+        "0.-1",
+        "+1.0",
+        "beef.0",
```

Comment:
Nitpick: maybe also test for something like "1.beef"?

Reply:
As we also have a max limit, is there any value in testing that larger ints than this get rejected?

Reply:
Please add a test for the length checks. at least It should check 
- maxLentgth - 1, 
- maxLentgh,
-  maxLength + 1. 
- Very big string does not overflow the converter. (In particular length check should be done first).

Reply:
Can be done another PR.